lib/public/components/Filters/common/filtersPanelPopover.js

```diff
@@ -46,7 +67,9 @@ const filtersToggleContent = (filteringModel, columns) => h('.w-l.scroll-y.flex-
     ]),
     h('.flex-column.g2', [
         Object.entries(columns)
-            .filter(([_, column]) => column.filter)
+            .filter(([_, column]) =>column.filter
+                && (applyProfile(column, appliedProfile, adaptColumnProfiles(column.profiles))
+                 || (column.filterProfiles ?? []).includes(appliedProfile)))
```

Comment:
Except if it's very clear, lambda should be on one line or have a body.
Here it would be way more clear to have:

However, I disagree with what you try to achieve here. The way profiles currently work is, if it's specified, profiles might disable columns or override their configuration. `filter` being one of the configuration properties, should be overridden in the profile property.
I agree that profiles system is flawed, and the whole active column system should be fixed, but adding this kind of dupplicated functionality can only make it worse.

Reply:
refactored